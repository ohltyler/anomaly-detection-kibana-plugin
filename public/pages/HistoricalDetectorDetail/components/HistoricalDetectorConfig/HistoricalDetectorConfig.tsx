/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import ContentPanel from '../../../../components/ContentPanel/ContentPanel';
import {
  //@ts-ignore
  EuiFlexGrid,
  EuiFlexItem,
  EuiText,
  EuiFormRow,
  EuiButton,
  EuiFormRowProps,
} from '@elastic/eui';
import { Detector } from '../../../../models/interfaces';
import React from 'react';
import moment from 'moment';

interface HistoricalDetectorConfigProps {
  detector: Detector;
  onEditDetector(): void;
}

const FixedWidthRow = (props: EuiFormRowProps) => (
  <EuiFormRow {...props} style={{ width: '250px' }} />
);

const renderCell = (title: string, description: string) => {
  return (
    <FixedWidthRow label={title}>
      <EuiText>
        <p className="enabled">{description}</p>
      </EuiText>
    </FixedWidthRow>
  );
};

const renderDate = (obj: any) => {
  if (typeof obj == 'number') {
    return moment(obj).format('MM/DD/YYYY hh:mm A');
  } else {
    return '-';
  }
};

export const HistoricalDetectorConfig = (
  props: HistoricalDetectorConfigProps
) => {
  return (
    <ContentPanel
      title="Historical detector configuration"
      titleSize="s"
      panelStyles={{ margin: '0px' }}
      actions={[<EuiButton onClick={props.onEditDetector}>Edit</EuiButton>]}
    >
      <EuiFlexGrid columns={3} gutterSize="l" style={{ border: 'none' }}>
        <EuiFlexItem>{renderCell('Name', props.detector.name)}</EuiFlexItem>
        <EuiFlexItem>{renderCell('ID', props.detector.id)}</EuiFlexItem>
        <EuiFlexItem>
          {renderCell(
            'Date range',
            renderDate(props.detector.detectionDateRange?.startTime) +
              '-' +
              renderDate(props.detector.detectionDateRange?.endTime)
          )}
        </EuiFlexItem>
        <EuiFlexItem>
          {renderCell('Description', props.detector.description)}
        </EuiFlexItem>
      </EuiFlexGrid>
    </ContentPanel>
  );
};
