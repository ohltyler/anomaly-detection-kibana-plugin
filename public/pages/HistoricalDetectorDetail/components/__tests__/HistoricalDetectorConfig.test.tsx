/*
 * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { render } from '@testing-library/react';
import { HistoricalDetectorConfig } from '../HistoricalDetectorConfig/HistoricalDetectorConfig';
import moment from 'moment';

const TITLE_TEXT = 'Historical detector configuration';
const TEST_DETECTOR = {
  id: 'test-id',
  name: 'test-name',
  detectionDateRange: {
    startTime: 0,
    endTime: 5,
  },
  description: 'test-description',
  lastUpdated: 0,
  indices: ['test-index'],
  detectionInterval: {
    period: {
      interval: 10,
      unit: 'Minutes',
    },
  },
};
const START_TIME_STRING = moment(0).format('MM/DD/YYYY hh:mm A');
const END_TIME_STRING = moment(5).format('MM/DD/YYYY hh:mm A');
const INTERVAL_STRING = '10 Minutes';
const mockOnEditDetector = jest.fn();

describe('<HistoricalDetectorConfig /> spec', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  describe('detector is not stopping', () => {
    test('renders the component', () => {
      const { container, getByText, queryByText } = render(
        <HistoricalDetectorConfig
          detector={TEST_DETECTOR}
          isStoppingDetector={false}
          onEditDetector={mockOnEditDetector}
        />
      );
      expect(container.firstChild).toMatchSnapshot();
      getByText(TITLE_TEXT);
      getByText(TEST_DETECTOR.id);
      getByText(TEST_DETECTOR.name);
      getByText(TEST_DETECTOR.description);
      getByText(TEST_DETECTOR.indices[0]);
      getByText(INTERVAL_STRING);
      getByText(START_TIME_STRING);
      getByText(START_TIME_STRING + '-' + END_TIME_STRING);
    });
  });
});
