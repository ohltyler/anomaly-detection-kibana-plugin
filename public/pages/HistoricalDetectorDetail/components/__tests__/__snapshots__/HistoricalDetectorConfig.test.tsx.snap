// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HistoricalDetectorConfig /> spec detector is not stopping renders the component 1`] = `
<div
  class="euiPanel euiPanel--paddingMedium"
  style="padding: 20px; margin: 0px;"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding: 0px;"
  >
    <div
      class="euiFlexItem"
    >
      <h3
        class="euiTitle euiTitle--small"
      >
        Historical detector configuration
      </h3>
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        />
      </div>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <button
            class="euiButton euiButton--primary"
            data-test-subj="editDetectorButton"
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Edit
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
  />
  <div
    style="padding: 10px 0px;"
  >
    <div
      class="euiFlexGrid euiFlexGrid--gutterLarge euiFlexGrid--thirds euiFlexGrid--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Name
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                test-name
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              ID
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                test-id
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Date range
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                12/31/1969 04:00 PM-12/31/1969 04:00 PM
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Description
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                test-description
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Last updated
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                12/31/1969 04:00 PM
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Data source index
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                test-index
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Detection interval
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                10 Minutes
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<HistoricalDetectorConfig /> spec detector is stopping renders the component 1`] = `
<div
  class="euiPanel euiPanel--paddingMedium"
  style="padding: 20px; margin: 0px;"
>
  <div
    class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
    style="padding: 0px;"
  >
    <div
      class="euiFlexItem"
    >
      <h3
        class="euiTitle euiTitle--small"
      >
        Historical detector configuration
      </h3>
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        />
      </div>
    </div>
    <div
      class="euiFlexItem euiFlexItem--flexGrowZero"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
      >
        <div
          class="euiFlexItem"
        >
          <button
            class="euiButton euiButton--primary euiButton-isDisabled"
            data-test-subj="editDetectorButton"
            disabled=""
            type="button"
          >
            <span
              class="euiButtonContent euiButton__content"
            >
              <span
                class="euiButton__text"
              >
                Edit
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <hr
    class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
  />
  <div
    style="padding: 10px 0px;"
  >
    <div
      class="euiFlexGrid euiFlexGrid--gutterLarge euiFlexGrid--thirds euiFlexGrid--responsive"
    >
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Name
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                test-name
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              ID
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                test-id
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Date range
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                12/31/1969 04:00 PM-12/31/1969 04:00 PM
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Description
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                test-description
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Last updated
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                12/31/1969 04:00 PM
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Data source index
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                test-index
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="euiFlexItem"
      >
        <div
          class="euiFormRow"
          id="random_html_id-row"
          style="width: 250px;"
        >
          <div
            class="euiFormRow__labelWrapper"
          >
            <label
              class="euiFormLabel euiFormRow__label"
              for="random_html_id"
            >
              Detection interval
            </label>
          </div>
          <div
            class="euiFormRow__fieldWrapper"
          >
            <div
              class="euiText euiText--medium"
              id="random_html_id"
            >
              <p
                class="enabled"
              >
                10 Minutes
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
